sandbox/trim-low-abund.py

```diff
@@ -130,36 +154,77 @@ def main():
 
         pass2list.append((filename, pass2filename, trimfilename))
 
+        screed_iter = screed.open(filename)
         pass2fp = open(pass2filename, 'w')
         trimfp = open(trimfilename, 'w')
 
         save_pass2 = 0
-        for n, read in enumerate(screed.open(filename)):
+        for n, is_pair, read1, read2 in broken_paired_reader(screed_iter):
             if n % 10000 == 0:
                 print '...', n, filename, save_pass2, read_reads, read_bp, \
                     wrote_reads, wrote_bp
 
-            read_reads += 1
-            read_bp += len(read.sequence)
+            # we want to track paired reads here, to make sure that pairs
+            # are not split between first pass and second pass.
 
-            seq = read.sequence.replace('N', 'A')
-            med, _, _ = ht.get_median_count(seq)
+            if is_pair:
+                read_reads += 2
+                read_bp += len(read1.sequence) + len(read2.sequence)
 
-            # has this portion of the graph saturated? if not,
-            # consume & save => pass2.
-            if med < NORMALIZE_LIMIT:
-                ht.consume(seq)
-                pass2fp.write(output_single(read))
-                save_pass2 += 1
-            else:                       # trim!!
-                trim_seq, trim_at = ht.trim_on_abundance(seq, CUTOFF)
-                if trim_at >= K:
-                    trimfp.write(output_single(read, trim_at))
-                    wrote_reads += 1
-                    wrote_bp += trim_at
-                    if trim_at != len(read.sequence):
+                seq1 = read1.sequence.replace('N', 'A')
+                seq2 = read2.sequence.replace('N', 'A')
+
+                med1, _, _ = ht.get_median_count(seq1)
```

Comment:
This is something I've seen in other code but still am not sure what it is--What are the underscores, here?